src/vmm/src/boot.rs

```diff
@@ -133,17 +131,139 @@ pub fn build_bootparams(
             add_e820_entry(
                 &mut params,
                 mmio_gap_end.raw_value(),
+                // The unchecked_offset_from is safe because:
+                // * mmio_gap_end > mmio_gap_start, validated at the beginning of the function
+                // * mmio_gap_start > himem_start, validated above
+                // therefore, mmio_gap_end > himem_start
```

Comment:
Isn't it enough to say the `unchecked_offset_from` is safe because last_addr > mmio_gap_end (true because of the above `if`)?